daos/models/users/user_model.py

```diff
@@ -1,13 +1,21 @@
 from dataclasses import dataclass
-from uuid import uuid4
+from sqlalchemy.orm import validates
+
+from daos.models.models_utils import auto_str, generate_uuid, validate_empty_string
 
 
 def define_user(db):
-    @dataclass
+
+    @auto_str
+    @dataclass(init=True, repr=True, eq=True)
     class User(db.Model):
-        __table_args__ = {'extend_existing': True}
-        id: str = db.Column(db.String(36), primary_key=True, default=str(uuid4()))
-        firstName: str = db.Column(db.String(250), nullable=False)
+        __table_args__ = {"extend_existing": True}
+        id: str = db.Column(db.String(36), primary_key=True, default=generate_uuid)
+        firstName: str = db.Column(db.String(250), nullable=False, unique=True)
         lastName: str = db.Column(db.String(250), nullable=False)
```

Comment:
can you please explain how it is related to the PR?
seems like more related to the user epic content